### PR TITLE
Add instrumentation compatibility documentation

### LINKED ISSUE #26: Document instrumentation versioning compatibility
For each support instrumented library document what version of the library are supported.

### PR DESCRIPTION
**Pull request by MrAlias**
This is manually generated. Automation of this process is left as a post-beta release task.

Resolve #26

### CODE REVIEW

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Instrumented Library Compatibility
+
+Instrumentation is provided for the following Go packages.
+
+- [database/sql](#database/sql)
+- [github.com/segmentio/kafka-go](#github.com/segmentio/kafka-go)
+- [google.golang.org/grpc](#google.golang.org/grpc)
+- [net/http](#net/http)
```
Comment (pellared) [inline COMPATIBILITY.md:8]:
```suggestion
- [`database/sql`](#database/sql)
- [`github.com/segmentio/kafka-go`](#github.com/segmentio/kafka-go)
- [`google.golang.org/grpc`](#google.golang.org/grpc)
- [`net/http`](#net/http)
```

File: COMPATIBILITY.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Instrumented Library Compatibility
+
+Instrumentation is provided for the following Go packages.
```
Comment (pellared) [inline COMPATIBILITY.md:3]:
```suggestion
Instrumentation is provided for the following Go libraries.
```

Why? Sometimes a library (module) consists of several packages (and even modules?). I think that "library" is a more general term and package and module are implementation specific. Also in OTel terminology we have "instrumented libraries". Related PR https://github.com/open-telemetry/opentelemetry.io/pull/4727

File: COMPATIBILITY.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+- [google.golang.org/grpc](#google.golang.org/grpc)
+- [net/http](#net/http)
+
+## database/sql
+
+[Package documentation](https://pkg.go.dev/database/sql)
+
+Supported version ranges: 
+
+- 1.12.0 to 1.22.5
```
Comment (pellared) [inline COMPATIBILITY.md:16]: I propose to use the same naming as in https://pkg.go.dev/database/sql?tab=versions

```suggestion
- `go1.12` to `go1.22.5`
```

File: COMPATIBILITY.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+## github.com/segmentio/kafka-go
+
+[Package documentation](https://pkg.go.dev/github.com/segmentio/kafka-go)
+
+Supported version ranges: 
+
+- v0.4.1 to v0.4.47
```
Comment (pellared) [inline COMPATIBILITY.md:24]:
```suggestion
- `v0.4.1` to `v0.4.47`
```

File: COMPATIBILITY.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## google.golang.org/grpc
+
+[Package documentation](https://pkg.go.dev/google.golang.org/grpc)
+
+Supported version ranges: 
+
+- v1.14.0 to v1.66.0
```
Comment (pellared) [inline COMPATIBILITY.md:32]:
```suggestion
- `v1.14.0` to `v1.66.0`
```

File: COMPATIBILITY.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+## net/http
+
+[Package documentation](https://pkg.go.dev/net/http)
+
+Supported version ranges: 
+
+- 1.12. to 1.22.5
```
Comment (pellared) [inline COMPATIBILITY.md:40]: I propose to use the same naming as in https://pkg.go.dev/net/http?tab=versions:

```suggestion
- `go1.12 to `go1.22.5`
```

**Review by pellared — APPROVED** [review]

File: COMPATIBILITY.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+- `v1.14.0` to `v1.66.0`
+
+## net/http
+
+[Package documentation](https://pkg.go.dev/net/http)
+
+Supported version ranges: 
+
+- `go1.12 to `go1.22.5`
```
Comment (MrAlias) [inline COMPATIBILITY.md:40]:
```suggestion
- `go1.12` to `go1.22.5`
```

File: COMPATIBILITY.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# Instrumented Library Compatibility
+
+Instrumentation is provided for the following Go libraries.
+
+- [`database/sql`](#database/sql)
+- [`github.com/segmentio/kafka-go`](#github.com/segmentio/kafka-go)
+- [`google.golang.org/grpc`](#google.golang.org/grpc)
+- [`net/http`](#net/http)
```
Comment (RonFed) [inline COMPATIBILITY.md:8]: Do we need to include the `otelglobal` instrumentation here? or maybe in a separate section about compatibility manual instrumentation.

Comment (MrAlias) [reply COMPATIBILITY.md:21]: Updated. PTAL.

File: COMPATIBILITY.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+[`otel`]: https://pkg.go.dev/go.opentelemetry.io/otel
+
+## Instrumented Library Compatibility
+
+Instrumentation is provided for the following Go libraries.
+
+- [`database/sql`](#database/sql)
+- [`github.com/segmentio/kafka-go`](#github.com/segmentio/kafka-go)
+- [`google.golang.org/grpc`](#google.golang.org/grpc)
+- [`net/http`](#net/http)
```
Comment (grcevski) [inline COMPATIBILITY.md:21]: Should we mention Gorilla Mux, since it's also working and it's a popular library?

Comment (MrAlias) [reply COMPATIBILITY.md:21]: We don't do any testing on what versions we support as far as I know.

I think we could add that, any maybe gin as well, as a follow up to this PR. We would need to develop a way to check what versions were actually supported though.

Comment (MrAlias) [reply COMPATIBILITY.md:21]: Captured in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/944

**Pull request merged**